File: ComposicionAgregacion/ejercicio 12/Hospital.java
import java.util.ArrayList;
public class Hospital {
    private String nombre;
    private ArrayList <Doctor> doctores;
    
    public Hospital(String nombre){
        this.nombre=nombre;
        this.doctores=new ArrayList<>();
    }
    public void agregarDoctor(Doctor doctor){
        doctores.add(doctor);
    }
    public void mostrarDoctores() {
    System.out.println("Doctores en " + nombre + ":");
    for (int i = 0; i < doctores.size(); i++) {
        System.out.println((i + 1) + ". " + doctores.get(i));
    }
    System.out.println();
}


}

File: ComposicionAgregacion/ejercicio 12/Hospital.py
class Doctor:
    def __init__(self, nombre, especialidad):
        self.nombre = nombre
        self.especialidad = especialidad

    def __str__(self):
        return f"{self.nombre} ({self.especialidad})"

class Hospital:
    def __init__(self, nombre):
        self.nombre = nombre
        self.doctores = []

    def agregar_doctor(self, doctor):
        self.doctores.append(doctor)

    def mostrar_doctores(self):
        print(f"Doctores en {self.nombre}:")
        for doctor in self.doctores:
            print(f"- {doctor}")
        print()

d1 = Doctor("Ana Pérez", "Cardiología")
d2 = Doctor("Luis Gómez", "Neurología")

h1 = Hospital("Hospital Central")
h2 = Hospital("Hospital Norte")

h1.agregar_doctor(d1)
h1.agregar_doctor(d2)
h2.agregar_doctor(d1)  

h1.mostrar_doctores()
h2.mostrar_doctores()
